packages/blocks/src/store/reducer.js

```diff
@@ -97,6 +97,47 @@ export function blockStyles( state = {}, action ) {
 	return state;
 }
 
+/**
+ * Reducer managing the block patterns.
+ *
+ * @param {Object} state  Current state.
+ * @param {Object} action Dispatched action.
+ *
+ * @return {Object} Updated state.
+ */
+export function blockPatterns( state = {}, action ) {
+	switch ( action.type ) {
+		case 'ADD_BLOCK_TYPES':
+			return {
+				...state,
+				...mapValues( keyBy( action.blockTypes, 'name' ), ( blockType ) => {
+					return uniqBy( [
+						...get( blockType, [ 'patterns' ], [] ),
+						...get( state, [ blockType.name ], [] ),
+					], ( style ) => style.name );
```

Comment:
I think this should read `( pattern ) => pattern.name` :)

Reply:
Oooops 😂

Reply:
Refactor planned in #18398 is going to fix it.